File: c/hamming/src/hamming.c
#include "hamming.h"

int compute(const char *a, const char *b)
{
    if (!a || !b)
    {
        return -1;
    }

    int distance = 0;

    int i = 0;
    for (; a[i] && b[i]; i++)
    {
        distance += (a[i] == b[i]) ? 0 : 1;
    }

    return (a[i] || b[i]) ? -1 : distance;
}

File: erlang/hamming/src/hamming.erl
-module(hamming).

-export([distance/2]).

distance(Str1, Str2) -> distance(Str1, Str2, 0).

distance([], [], Counter) -> Counter;
distance([H | T1], [H | T2], Counter) -> distance(T1, T2, Counter);
distance([_ | T1], [_ | T2], Counter) -> distance(T1, T2, Counter + 1);
distance(_, _, _) -> {error, "left and right strands must be of equal length"}.
